modules/caddyhttp/replacer.go

```diff
@@ -398,6 +403,20 @@ func getTLSPeerCert(cs *tls.ConnectionState) *x509.Certificate {
 	return cs.PeerCertificates[0]
 }
 
+type requestId struct {
+	value string
+}
+
+// Lazy generates UUID string or return cached value if present
+func (rid *requestId) String() string {
+	if rid.value == "" {
+		if id, err := uuid.NewRandom(); err == nil {
+			rid.value = id.String()
```

Comment:
Actually, I'm pretty sure this is racey. Can you test this with multiple requests using `go build -race` (or `go run -race`)?

Reply:
It's not racey. I tested with `go build -race` and `wrk`.

Reply:
Ah, of course. I'm an idiot... these values are only used within this request, which is generally all on a single goroutine, so it's not a race. 🤦‍♂️